graders/python/python_autograder/code_feedback.py

```diff
@@ -299,6 +299,59 @@ def bad(msg):
 
         return True
 
+    @classmethod
+    def check_dict(
+        cls,
+        name,
+        ref,
+        data,
+        entry_type=None,
+        accuracy_critical=False,
+        report_failure=True,
+    ):
+        """
+        Feedback.check_dict(name, ref, data)
+
+        Check that a student dict has correct length with respect to a reference dict.  Can also check for a homogeneous data type for the list.
+
+        - ``name``: Name of the dictionary that is being checked. This will be used to give feedback.
+        - ``ref``: Reference dictionary.
+        - ``data``: Student dictionary to be checked.
+        - entry_type: If not None, checks if each value in the student dictionary has this type.
+        - ``accuracy_critical``: If true, grading will halt on failure.
+        - ``report_failure``: If true, feedback will be given on failure.
```

Comment:
Some more parameters would be useful, like checking keys and values or just values only, and maybe even checking that the student answer contains some keys in particular instead of just equality.

Reply:
I have added:
- ``partial_keys``: If not None, it takes a List of keys to check if these particular keys are present in the student's dict or not.
- ``check_only_values``: If true, grading will be done only based on checking all valeus in student's dict and reference's dict match or not.
- ``check_only_keys``: If true, grading will be done only based on checking all keys in student's dict and reference's dict match or not.

Reply:
- ``entry_type_key``: If not None, requires that each key in the student's dictionary in solution be of this type.
- ``entry_type_value``: If not None, requires that each value in the student's dictionary in solution be of this type.